parsec/core/gui/new_version.py

```diff
@@ -28,18 +26,49 @@ def _extract_version_tuple(raw):
         return None
 
 
-async def _do_check_new_version(url):
-    # urlopen automatically follows redirections
-    resolved_url = await trio.to_thread.run_sync(
-        lambda: urlopen(Request(url, method="HEAD")).geturl()
-    )
+def _tuple_to_version_number(version):
+    return ".".join([str(i) for i in version])
+
+
+async def _async_get(url, method="GET"):
+    return await trio.to_thread.run_sync(lambda: urlopen(Request(url, method=method)))
 
-    lastest_version = _extract_version_tuple(resolved_url)
+
+async def _do_check_new_version(url, api_url):
+    # urlopen automatically follows redirections
+    resolved_url = (await _async_get(url, method="HEAD")).geturl()
+    latest_from_head = _extract_version_tuple(resolved_url)
     current_version = _extract_version_tuple(__version__)
-    if lastest_version and current_version and current_version < lastest_version:
-        return lastest_version
-    else:
-        return None
+    if latest_from_head and current_version and current_version < latest_from_head:
+        json_releases = json.loads((await _async_get(api_url)).read())
+
+        current_arch = QSysInfo().currentCpuArchitecture()
```

Comment:
We've been using `is_64bits = sys.maxsize > 2 ** 32` in winfspy, but this may be cleaner if Qt is available...